AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/OrganizationControllerTests.cs

```diff
@@ -28,246 +30,291 @@ public class OrganizationControllerTests
-
-                var controller = new OrganizationController(mockMediator.Object, SuccessValidator());
-                var result = (RedirectToActionResult) await controller.Edit(model);
-
-          Assert.Equal("Details", result.ActionName);
-          Assert.Equal("Admin", result.RouteValues["area"]);
-          Assert.Equal(Id, result.RouteValues["id"]);
-        }
-
-            [Fact(Skip = "NotImplemented")]
-            public void EditPostAddsErrorToModelState_WhenModelStateIsValid_AndOrganizationNameIsNotUnique()
-            {
-            }
-
-            [Fact]
-            public void EditPostShouldHaveValidateAntiForgeryTokenAttribute()
-            {
-                MethodShouldHaveValidateAntiForgeryTokenAttribute("Edit", typeof(OrganizationEditViewModel));
-            }
-
-            [Fact]
-            public void EditPostShouldHaveHttpPostAttribute()
-            {
-                MethodShouldHaveHttpPostAttribute("Edit", typeof(OrganizationEditViewModel));
-            }
-
-            #endregion
-
+
+        [Fact]
+        public async Task EditGetSendsOrganizationEditQueryWithCorrectOrganizationId()
+        {
+            const int organizationId = 1;
+            var mediator = new Mock<IMediator>();
+
+            var sut = new OrganizationController(mediator.Object, null);
+            await sut.Edit(organizationId);
+
+            mediator.Verify(x => x.SendAsync(It.Is<OrganizationEditQuery>(y => y.Id == organizationId)), Times.Once);
+        }
+
+        [Fact]
+        public async Task EditGetReturnsHttpNotFoundResult_WhenOrganizationIsNotFound()
+        {
+            var mediator = new Mock<IMediator>();
+            mediator.Setup(x => x.SendAsync(It.IsAny<OrganizationEditQuery>())).ReturnsAsync(null);
+
+            var sut = new OrganizationController(mediator.Object, null);
+            var result = await sut.Edit(It.IsAny<int>());
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async Task EditGetReturnsCorrectViewAndViewModel_WhenMediatorReturnsOrganizationEditModel()
+        {
+            CreateSut();
+
+            var model = new OrganizationEditViewModel();
+
+            _mediator.Setup(x => x.SendAsync(It.Is<OrganizationEditQuery>(y => y.Id == Id))).ReturnsAsync(model);
+
+            var result = (ViewResult)await _sut.Edit(Id);
+
+            Assert.Equal("Edit", result.ViewName);
+            Assert.Same(model, result.ViewData.Model);
+        }
+
+        [Fact]
+        public async Task EditPostReturnsBadRequestResult_WhenModelIsNull()
+        {
+            var sut = new OrganizationController(null, null);
+            var result = await sut.Edit(null);
+            Assert.IsType<BadRequestResult>(result);
+        }
+
+        [Fact]
+        public async Task EditPostInvokesOrganizationValidatorsValidateMethodWithTheCorrectOrganizationEditModel_WhenModelIsNotNull()
+        {
```

Comment:
Thanks Michael, I appreciate your recommendations, I have one question, when the test says `TheCorrectOrganizationEditModel` what exactly constitutes a correct EditModel? is it the virtue of it being the same type or the same instance? If Instead of `It.IsAny<OrganizationEditViewModel>` I new up an instance and pass that, would it be sufficient proof that our ViewModel is correct? Thanks